#! /bin/sh -e

# DP: distutils: Add an option --install-layout=deb, which is ignored for 2.4.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/distutils/command/install.py~	2008-12-20 15:55:55.000000000 +0100
+++ Lib/distutils/command/install.py	2008-12-20 16:05:29.000000000 +0100
@@ -144,6 +144,9 @@
 
         ('record=', None,
          "filename in which to record list of installed files"),
+
+        ('install-layout=', None,
+         "installation layout to choose (known values: deb)"),
         ]
 
     boolean_options = ['compile', 'force', 'skip-build']
@@ -177,6 +180,9 @@
         self.install_scripts = None
         self.install_data = None
 
+        # enable custom installation, known values: deb, IGNORED in 2.4
+        self.install_layout = None
+
         self.compile = None
         self.optimize = None
 
